### ISSUE TITLE
Style/RegexpLiteral not accounting for ambigous first argument

### ISSUE THREAD

**Issue by mojavelinux** [rubocop/rubocop#9874]
If we enable the Style/RegexpLiteral rule together with Style/MethodCallWithArgsParentheses using the EnforcedStyle omit_parentheses, RuboCop incorrectly recommends using // around a regular expression when the %r style is needed to avoid an ambiguous first argument.

## Expected behavior

Consider the following rules:

```yaml
Style/MethodCallWithArgsParentheses:
  Enabled: true
  EnforcedStyle: omit_parentheses
Style/RegexpLiteral:
  Enabled: false
```

Now consider the following code:

```rb
str = 'foo'
str = str.gsub %r/foo/, 'bar'
puts str
```

We expect RuboCop to be fine with this code because %r is needed to avoid an ambiguous first argument warning from Ruby.

## Actual behavior

RuboCop reports the following problem:

```
C: [Correctable] Style/RegexpLiteral: Use // around regular expression.
```

If we take away the %r, Ruby reports the following warning:

```
warning: ambiguous first argument; put parentheses or a space even after `/' operator
```

I think RuboCop should recognize this situation and allow for the use of %r in this case.

## RuboCop version

```
1.16.0 (using Parser 3.0.1.1, rubocop-ast 1.7.0, running on ruby 2.7.2 x86_64-linux)
  - rubocop-rake 0.5.1
  - rubocop-rspec 2.4.0
```
